### PR TITLE
[INF] Require pyspark minimal version is v3.2.0 to cut duplicates codes

### LINKED ISSUE #1110: Require pyspark minimal version is v3.2.0 to cut duplicates codes
Since [pyspark v3.2.0](https://github.com/apache/spark/blob/5d45a415f3a29898d92380380cfd82bfc7f579ea/python/pyspark/pandas/extensions.py#L28-L64), it has contained `CachedAccessor`, `_register_accessor`, `_register_accessor`

janitor requires pyspark minimal version is v3.1.2 at present.
Compared to v3.1.2, v3.2.0 is a minor version.

https://github.com/pyjanitor-devs/pyjanitor/blob/4867b20273e99cb2f4c32278123d5c23c2ccaa67/janitor/spark/backend.py#L9-L37

Note: The pyspark in the [setup.py](https://github.com/pyjanitor-devs/pyjanitor/blob/4867b20273e99cb2f4c32278123d5c23c2ccaa67/.requirements/spark.txt#L9) file requires v3.2.1 but ci ([environment-dev.yml](https://github.com/pyjanitor-devs/pyjanitor/blob/4867b20273e99cb2f4c32278123d5c23c2ccaa67/environment-dev.yml#L41)) requires v3.1.2.

### PR DESCRIPTION
**Pull request by Zeroto521**
# PR Description

Please describe the changes proposed in the pull request:

Directly import `register_dataframe_accessor` from pyspark

**This PR resolves #1110.**

# PR Checklist

Please ensure that you have done the following:

1. [x] PR in from a fork off your branch. Do not PR from `<your_username>`:`dev`, but rather from `<your_username>`:`<feature-branch_name>`.

2. [x] If you're not on the contributors list, add yourself to `AUTHORS.md`.

3. [x] Add a line to `CHANGELOG.md` under the latest version header (i.e. the one that is "on deck") describing the contribution.
    - Do use some discretion here; if there are multiple PRs that are related, keep them in a single line.

# Automatic checks

There will be automatic checks run on the PR. These include:

- Building a preview of the docs on Netlify
- Automatically linting the code
- Making sure the code is documented
- Making sure that all tests are passed
- Making sure that code coverage doesn't go down.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

File: environment-dev.yml [hunk 1]
```diff
@@ -38,7 +38,7 @@ dependencies:
   - pre-commit
   - pyflakes
   - pylint
-  - pyspark=3.1.2
+  - pyspark
```
Comment (Zeroto521) [inline environment-dev.yml:41]: Install the latest pyspark version

File: environment-dev.yml [hunk 2]
```diff
@@ -38,7 +38,7 @@ dependencies:
   - pre-commit
   - pyflakes
   - pylint
-  - pyspark=3.1.2
+  - pyspark>=3.2.0
```
Comment (Zeroto521) [inline environment-dev.yml:41]: Install the latest version without conflict.

**Review by ericmjl — APPROVED** [review]
Given the code deletions, I'm not worried about the code coverage. I approve!

**Comment by Zeroto521** [comment]
A solution to handle coverage is to build different testing envs.

The fuzzy version of dependencies:
1. pyjanitor requires the minimal python version is 3.6, but the python version in testing env is [3.9](https://github.com/pyjanitor-devs/pyjanitor/blob/dev/environment-dev.yml)
2. pyanitor requires pandas 1.4.1 (in [.requirements/base.txt](https://github.com/pyjanitor-devs/pyjanitor/blob/63c075ec935493faf0ca8aa19e8bd1c601258e7c/.requirements/base.txt#L18)), but pandas 1.4.1 requires minimal the python version is 3.8
3. The CI only test the specific version of dependencies.

Maybe we could set a series of envs to test pyjanitor
- 310-minimal: minimal version of dependencies under python 3.10
- 39-minimal: minimal version of dependencies under python 3.9
- 38-minimal: minimal version of dependencies under python 3.8
- latest: latest version of dependencies with python from 3.8 to 3.10 without '=='

**Comment by ericmjl** [comment]
That sounds like a good idea. Let's take that to a different PR, though, so this one can get closed out earlier than later.

**Comment by Zeroto521** [comment]
Agree, I will merge this PR in later.
The multi-testing environments should be put in progress before releasing v1.0.

**Pull request merged**